workorders.lic

```diff
@@ -286,28 +286,63 @@ class WorkOrders
       stock_needed = 1
       order_rem(info['stock-room'], stock_needed, info['stock-number'])
     when 'You get'
-      /are (.*) parts left of the water/ =~ bput('count my water', 'are .* parts left of the water')
+      /are (.*) parts left of the water|only (.*) part left/ =~ bput('count my water', 'are .* parts left of the water')
       existing = Regexp.last_match(1).scan(/\w+/).to_a
+      /is (.*) part left of the water/ =~ bput('count my water', 'only .* part left')
+      existing = Regexp.last_match(1).scan(/\w+/).to_a    
       val_map = { 'one' => 1, 'two' => 2, 'three' => 3, 'four' => 4, 'five' => 5, 'six' => 6, 'seven' => 7, 'eight' => 8, 'nine' => 9, 'ten' => 10, 'eleven' => 11, 'twelve' => 12, 'thirteen' => 13, 'fourteen' => 14, 'fifteen' => 15, 'sixteen' => 16, 'seventeen' => 17, 'eighteen' => 18, 'nineteen' => 19, 'twenty' => 20, 'thirty' => 30, 'fourty' => 40, 'fifty' => 50, 'sixty' => 60, 'seventy' => 70, 'eighty' => 80, 'ninety' => 90 }
-      value = existing.map { |word| val_map[word] }.inject(&:+)
+      value = water.map { |word| val_map[word] }.inject(&:+)
       stock_needed = ((quantity * recipe['volume'] - value) / 100.0).ceil
     end
 
     if stock_needed < 3
       order_rem(info['stock-room'], stock_needed, info['stock-number'])
+      bput("get water from #{@bag}", "You get", "What were")
+      bput("get water from #{@bag}", "You get", "What were")
+      bput('combine water', 'You combine')
       stow_tool('water')
     end
-
+    stow_tool('water')
     # order herbs
-
     order_rem(info['stock-room'], quantity, recipe['herb1_stock'])
-    if recipe['herb'] = !'na'
-      order_rem(info['stock-room'], stock_needed, recipe['herb2_stock'])
+#    if not recipe['herb2'].empty?
+     if recipe['herb2'] != 'na'
+      case bput("tap first #{recipe['herb2']} in my #{@bag}", "You tap", "I could not find")
+      when 'I could not find'
+        herb2_needed = 1
+        order_rem(info['stock-room'], herb2_needed, recipe['herb2_stock'])
+      when 'You tap'
+        case bput("tap second #{recipe['herb2']} in my #{@bag}", "You tap", "I could not find")
+           when 'You tap'
+             bput("get #{recipe['herb2']} from my #{@bag}", "You get")
+             bput("get #{recipe['herb2']} from my #{@bag}", "You get")
+             case bput('combine', 'You combine', 'That stack of herbs')
+               when 'That stack of herbs'
+                 stow_tool(recipe['herb2'])
+                 stow_tool(recipe['herb2'])
+               when 'You get'
+                 case bput("count #{recipe['herb2']}", "You count out \d+ pieces").scan(/\d+/).first.to_i
+                   when /You count out (\d+) pieces/
+                     herb2_have = Regexp.last_match(1)
+                     if herb2_have < quantity 
```

Comment:
I don't see a corresponding `end` for this `if`, nor are the line under it indented. I think the `end` on line 340 is making this work in an unintended way.